core/src/main/java/org/jboss/jandex/Indexer.java

```diff
@@ -2557,7 +2622,11 @@ public Index complete() {
         propagateTypeVariables();
 
         try {
-            return new Index(masterAnnotations, subclasses, subinterfaces, implementors, classes, modules, users);
+            Map<DotName, List<ClassInfo>> userLists = new HashMap<>();
+            for (Map.Entry<DotName, Set<ClassInfo>> entry : users.entrySet()) {
+                userLists.put(entry.getKey(), new ArrayList<>(entry.getValue()));
```

Comment:
I wonder if `List#copyOf()` could be a little bit more efficient here?

Reply:
Possibly, but Jandex is still on Java 8 :-)

Reply:
Ah, sorry. I forgot about it :facepalm: